third_party/ienn/src/ie_compilation.cpp

```diff
@@ -352,14 +376,38 @@ int32_t Compilation::AddConvolution(const Operation& operation) {
         params.nhwc_layout ? TransposeLayout(index_op_map_[input_index], true)
                            : index_op_map_[input_index];
     if (params.depthwise) {
-      SizeVector weights_dims{params.depth_out, 1, 1, params.filter_height,
-                              params.filter_width};
-      AddConstant(weights_index, params.nhwc_layout, weights_dims);
+      if (index_op_map_.find(weights_index) == index_op_map_.end()) {
+        // Setup constants
+        SizeVector weights_dims{params.depth_out, 1, 1, params.filter_height,
+                                params.filter_width};
+        result = AddConstant(weights_index, params.nhwc_layout, weights_dims);
+        if (result != error_t::NOT_ERROR) {
+          return result;
+        }
+      } else {
+        if (params.nhwc_layout) {
+          // transpose the weights from nghwc to ngchw
+          index_op_map_[weights_index] =
+              TransposeLayout(index_op_map_[weights_index], false, true);
+        }
+      }
       conv_node = std::make_shared<op::v1::GroupConvolution>(
           conv_input_node, index_op_map_[weights_index], Strides(strides),
           CoordinateDiff(padBegin), CoordinateDiff(padEnd), Strides(dilations));
     } else {
-      AddConstant(weights_index, params.nhwc_layout);
+      if (index_op_map_.find(weights_index) == index_op_map_.end()) {
+        // Setup constants
+        result = AddConstant(weights_index, params.nhwc_layout);
+        if (result != error_t::NOT_ERROR) {
+          return result;
+        }
+      } else {
+        if (params.nhwc_layout) {
+          // transpose the weights from nhwc to nchw
+          index_op_map_[weights_index] =
+              TransposeLayout(index_op_map_[weights_index], true);
```

Comment:
Does this also need to transpose the weights from nghwc to ngchw?

Reply:
This section code is aimed for common conv(not depthwise), the rank of weights are all 4D. So we need to transpose them from nhwc to nchw.